src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+    private Figure figure;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int index = random.nextInt(5);
+        switch (index) {
+            case 0:
+                figure = new Circle(colorSupplier.getRandomColor(), random.nextDouble());
+                break;
```

Comment:
```suggestion
                 return getRandomCircle();
```
lets add private methods to get each figure - this would make your code easier to read